src/ZeldaWindWaker/m_do_ext.ts

```diff
@@ -85,6 +101,221 @@ export class mDoExt_bvaAnm extends mDoExt_baseAnm<VAF1> {
+        this.material.allocateDrawParamsDataOnInst(renderInst, drawParams);
+        this.material.setOnRenderInst(renderInstManager.gfxRenderCache, renderInst);
+
+        renderInstManager.submitRenderInst(renderInst);
+    }
+
+    public updateWithScale(globals: dGlobals, segmentCount: number, scale: number, color: Color, space: number, tevStr: dKy_tevstr_c): void {
+        this.color = color;
+        this.tevStr = tevStr;
+        this.numSegments = Math.min(segmentCount, this.maxSegments);
+        const spacing = (space !== 0) ? scale / space : 0.0;
+
+        let dist = 0.0;
+        for (let i = 0; i < this.numLines; i++) {
+            const line = this.lines[i];
+            const srcPos = line.segments;
+            let segIdx = 0;
+            let vertIdx = 0;
+            assert(!!line.texCoords);
+
+            const dstPos = line.positions;
+            const dstUvs = line.texCoords;
+            let r_scale = scale;
+
+            // Handle the first segment
+            dstUvs[vertIdx + 0][1] = dist;
+            dstUvs[vertIdx + 1][1] = dist;
+
+            const segVec = vec3.sub(scratchVec3a, srcPos[segIdx + 1], srcPos[segIdx + 0]);
+            const delta = vec3.length(segVec);
+            dist += delta * 0.1;
+
+            // Normalize and then scale a vector orthogonal to both the segment and eye
+            const eyeVec = vec3.sub(scratchVec3b, srcPos[segIdx + 0], globals.camera.cameraPos);
+            const eyeCross = vec3.cross(scratchVec3a, segVec, eyeVec);
+            const mag = vec3.length(eyeCross);
+            if (mag !== 0.0) {
+                vec3.scale(eyeCross, eyeCross, scale / mag);
+            }
+
+            vec3.add(dstPos[vertIdx + 0], srcPos[segIdx], eyeCross);
+            vec3.sub(dstPos[vertIdx + 1], srcPos[segIdx], eyeCross);
+
+            segIdx += 1;
+            vertIdx += 2;
+            const nextP0 = vec3.add(scratchVec3c, srcPos[segIdx], eyeCross);
+            const nextP1 = vec3.sub(scratchVec3d, srcPos[segIdx], eyeCross);
+
+            // Handle all of the middle segments
+            for (let j = this.numSegments - 2; j > 0; j--) {
+                if (j < space) {
+                    r_scale -= spacing;
+                }
+
+                dstUvs[vertIdx + 0][1] = dist;
+                dstUvs[vertIdx + 1][1] = dist;
+
+                const segVec = vec3.sub(scratchVec3a, srcPos[segIdx + 1], srcPos[segIdx + 0]);
+                const delta = vec3.length(segVec);
+                dist += delta * 0.1;
+
+                // Normalize and then scale a vector orthogonal to both the segment and eye
+                const eyeVec = vec3.sub(scratchVec3b, srcPos[segIdx + 0], globals.camera.cameraPos);
+                const eyeCross = vec3.cross(scratchVec3a, segVec, eyeVec);
+                let mag = vec3.length(eyeCross);
+                if (mag !== 0.0) {
+                    mag = scale / mag;
+                }
+                vec3.scale(eyeCross, eyeCross, mag);
```

Comment:
I suspect this is supposed to be r_scale here in the inner loop?

```suggestion
                normalizeToLength(eyeCross, eyeCross, r_scale);
```